dnsutil/zone.go

```diff
@@ -0,0 +1,209 @@
+package dnsutil
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+
+	"github.com/miekg/dns"
+)
+
+type zoneGenData struct {
+	Origin     string
+	DefaultTtl uint32
+	Records    []dns.RR
+}
+
+func (z *zoneGenData) Len() int      { return len(z.Records) }
+func (z *zoneGenData) Swap(i, j int) { z.Records[i], z.Records[j] = z.Records[j], z.Records[i] }
+func (z *zoneGenData) Less(i, j int) bool {
+	a, b := z.Records[i], z.Records[j]
```

Comment:
Can we give it a sort.Sort like parameter that sorts for you?

Don't really like magical flags for this...